src/Microsoft.Diagnostics.Monitoring/Metrics/MetricsStore.cs

```diff
@@ -97,7 +97,11 @@ public async Task SnapshotMetrics(Stream outputStream, CancellationToken token)
             Dictionary<MetricKey, Queue<Metric>> copy = null;
             lock (_allMetrics)
             {
-                copy = new Dictionary<MetricKey, Queue<Metric>>(_allMetrics);
+                copy = new Dictionary<MetricKey, Queue<Metric>>();
```

Comment:
Another option would be to do this with Concurrent classes and bypass the copy altogether. Because metrics are limited in size I don't think it's something we have to do right now.

Reply:
While that would avoid the exception, that can lead to inconsistencies in the reported data.

For example, the remainder of the code in this method does at least two enumerations of each queue, so if something is inserted into the queue between each operation, you'll get different results. However, in this very specific case, I don't think that'll yield anything terribly wrong since the first enumeration is just getting the name of the first item (which should be the same name of all of the items in the queue).

Rather than having the remainder of the code be cognizant of the concurrency, just create the snapshot upfront.